Assets/Scripts/Entities/Containers/Inventory.cs

```diff
@@ -65,28 +72,63 @@ public bool add(Item newItem)
             return true;
         }
 
+        /// <summary>
+        /// Removes item from invetory
+        /// </summary>
+        /// <param name="i"></param>
+        /// <returns></returns>
+        public bool remove(int i)
+        {
+            if (i < 0 || i >= items.Count)
+            {
+                return false;
+            }
+
+            items[i] = null;
+
+            return true;
+        }
+
         /// <summary>
         /// Checks to see if item is <see cref="Useable"/>, then executes its <c>use()</c> method.
         /// </summary>
         /// <param name="i"> The element of inventory that will be used</param>
+        /// <param name="isConsumeable">is set if used item is consumeable</param>
         /// <returns>
         /// <list type="bullet"> 
         /// <item> true - The item was useable and its use fucntion was executed</item>
         /// <item> false - The item is not useable</item>
         /// </list>
         /// </returns>
-        public bool use(int i)
+        public bool use(int i, Combatant target, out bool isConsumeable)
```

Comment:
does isConsumeable need to be passed in here? you check for it in the method I think

Reply:
This `use()` method  **sets  isConsumable** back to the caller as an `out` parameter.

I do this as the return value of `use()` is to inform the caller whether the item is useable, and whether to perform amy necessary UI updates based on the use effect of the item. 

Additionally we want to know whether the used item should be consumed, to also perform any UI updates, hence the `out` reference type.

In effect  `out bool isConsumable` acts an additional return value of `use`.

**For more context**
The value of `isConsumable`  Is used to tell the UI whether to destroy/redraw the visual component of the item in the player's inventory.  

I could obtain the value of `isConsumable` from the internal state of the internal `items`  list. although the objective here is to propogate the value of isConsumable back to the UI layer. 

Perhaps I should decouple the checking of  **using** and **consuming** of an item for clarity sake. However I want to ensure that using of an item performs the necessary operation for consuming it, if required by its type